File: 기출/Naver Webtoon/2021공채/1.py
# 중복순열 + 슬라이딩 윈도우
from itertools import permutations

def solution(N, money, T, K):
    _range = []
    for i in range(N+1):
        _range += [i] * N
    candidates = set([perm for perm in permutations(_range, N) if sum(perm) == money])
    answer = 0
    testing = []
    for c in candidates:
        left, right = 0, T
        valid = True
        while right <= len(c):
            if sum(c[left:right]) > K:
                valid = False
                break
            left += 1
            right += 1
        if valid:
            answer += 1
            testing.append(c)
    print(testing)
    return answer
    

N = 4
money = 7
T = 2
K = 4

print(solution(N, money, T, K))

File: 기출/Naver Maps Internship 2021/1.swift

func calculateDigitSum(of n: Int) -> Int {
    var sum = 0
    var number = n
    while number != 0 {
        sum = sum + number % 10
        number = number / 10
    }
    return sum
}

public func solution(_ N : Int) -> Int {

    var number = 1
    var answer = 0
    while true {
        if calculateDigitSum(of: number) == N {
            answer = number
            break
        }
        number += 1
    }
    return answer
}
